spotbugs/src/main/java/edu/umd/cs/findbugs/sarif/SarifBugReporter.java

```diff
@@ -17,29 +19,42 @@ public SarifBugReporter(Project project) {
     @Override
     public void finish() {
         try {
-            JSONWriter jsonWriter = new JSONWriter(outputStream);
-            jsonWriter.object();
-            jsonWriter.key("version").value("2.1.0").key("$schema").value(
+            JsonWriter jsonWriter = new JsonWriter(outputStream);
+            jsonWriter.beginObject();
+            jsonWriter.name("version").value("2.1.0").name("$schema").value(
                     "https://raw.githubusercontent.com/oasis-tcs/sarif-spec/master/Schemata/sarif-schema-2.1.0.json");
             processRuns(jsonWriter);
             jsonWriter.endObject();
             getBugCollection().bugsPopulated();
+        } catch (IOException e) {
+            e.printStackTrace(System.err);
         } finally {
             outputStream.close();
         }
     }
 
-    private void processRuns(@NonNull JSONWriter jsonWriter) {
-        jsonWriter.key("runs").array().object();
-        BugCollectionAnalyser analyser = new BugCollectionAnalyser(getBugCollection());
-        processTool(jsonWriter, analyser.getRules());
-        processInvocations(jsonWriter, analyser.getBaseToId());
-        jsonWriter.key("results").value(analyser.getResults());
-        jsonWriter.key("originalUriBaseIds").value(analyser.getOriginalUriBaseIds());
-        jsonWriter.endObject().endArray();
+    private void processRuns(@NonNull JsonWriter jsonWriter) throws IOException {
+        try {
+            Gson gson = new Gson();
+            jsonWriter.name("runs").beginArray().beginObject();
+            BugCollectionAnalyser analyser = new BugCollectionAnalyser(getBugCollection());
+            processTool(jsonWriter, analyser.getRules());
+            processInvocations(jsonWriter, analyser.getBaseToId());
+
+            jsonWriter.name("results").beginArray();
+            analyser.getResults().forEach((result) -> gson.toJson(result, jsonWriter));
+            jsonWriter.endArray();
+
+            jsonWriter.name("originalUriBaseIds");
+            gson.toJson(analyser.getOriginalUriBaseIds(), jsonWriter);
+
+            jsonWriter.endObject().endArray(); // end "runs", end "runs" array
+        } catch (IOException e) {
+            throw e;
+        }
```

Comment:
This try-block seems needless.

```suggestion
        Gson gson = new Gson();
        jsonWriter.name("runs").beginArray().beginObject();
        BugCollectionAnalyser analyser = new BugCollectionAnalyser(getBugCollection());
        processTool(jsonWriter, analyser.getRules());
        processInvocations(jsonWriter, analyser.getBaseToId());

        jsonWriter.name("results").beginArray();
        analyser.getResults().forEach((result) -> gson.toJson(result, jsonWriter));
        jsonWriter.endArray();

        jsonWriter.name("originalUriBaseIds");
        gson.toJson(analyser.getOriginalUriBaseIds(), jsonWriter);

        jsonWriter.endObject().endArray(); // end "runs", end "runs" array
```